src/main/java/emissary/server/mvc/adapters/RequestUtil.java

```diff
@@ -73,6 +74,41 @@ public static String[] getParameterValues(final ServletRequest request, final St
         return retArray;
     }
 
+    /**
+     * Get attribute or parameters from request and return a list of Strings. Attribute has priority over parameter when
+     * both are present
+     */
+    public static List<String> getParameterValuesStringList(final ServletRequest request, final String param) {
+        List<String> retList = new ArrayList<>();
+        Object o = request.getAttribute(param);
+
+        if (o == null) {
+            try {
+                o = sanitizeParametersStringList(Arrays.asList(request.getParameterValues(param)));
+            } catch (NullPointerException e) {
+                logger.debug("RequestUtil.getParameterValues for {} is null", param);
+                return retList;
+            }
+        }
+
+        try {
+            retList = (List<String>) o;
+        } catch (ClassCastException e) {
+            retList = new ArrayList<>(1);
+            retList.add(0, (String) o);
+        }
+
+        if (retList != null) {
+            for (int i = 0; i < retList.size(); i++) {
+                logger.debug("RequestUtil.getParameterValues for {} [{}]: {}", param, i, retList.get(i));
+            }
+        } else {
+            logger.debug("RequestUtil.getParameterValues for {} is null", param);
```

Comment:
Would it be more helpful for this log (and possibly the one on line 89) to indicate `"RequestUtil.getParameterValuesStringList for {} [{}]: {}"` instead?